src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -400,6 +419,21 @@ private String getSamlRequestQueryStringSignature(String samlRequestQueryString)
         }
     }
 
+    private static Element getMetadataDOM(final String xmlString) throws IOException, SAXException, ParserConfigurationException {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
```

Comment:
Do we need to create DocumentBuilderFactory everytime getMetadataDOM is called ?

Reply:
We don't need a new DocumentBuilderFactory, but getMetadataDOM will be call only if user wants to use `metadata_body` to configure SAML auth. This will happen only on initializing/updating securityconfig which I don't think will happen very frequently.

Reply:
Okay. We can move DocumentBuilderFactory and setNamespaceAware to a static initializer ?